src/main/java/com/epam/onlinestore/repository/impl/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.epam.onlinestore.repository.impl;
+
+import com.epam.onlinestore.entity.Product;
+import com.epam.onlinestore.repository.ProductRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * a Layer class that contains methods for working with a database with a Product
+ */
+@Slf4j
+@Component
+public class ProductRepositoryImpl implements ProductRepository {
+    private final List<Product> productList = new ArrayList<>();
+
+    public Product getProductsById(long productId) {
+        return productList.stream()
+                .filter(product -> product.getId() == productId)
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Product is not found!"));
```

Comment:
I would recommend to create specific custom exception, like ProductNotFoundException instead of RuntimeException